.github/workflows/lint.yml

```diff
@@ -52,10 +52,11 @@ jobs:
         working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}
         run: |
           git fetch origin main
-          CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRTUXB origin/main | grep -E "\.(js|ts|tsx)$")
+          CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRTUXB origin/main | grep -E "\.(js|ts|tsx)$" || true)
           if [ -n "$CHANGED_FILES" ]; then
             echo "Linting changed files..."
             yarn lint $CHANGED_FILES
```

Comment:
```suggestion
          CHANGED_FILES=($(git diff --name-only --diff-filter=d origin/main | grep -E "\.(js|ts|tsx)$")) || true
          if [ -n "$CHANGED_FILES" ]; then
            echo "Linting changed files..."
            yarn lint "${CHANGED_FILES[@]}"
```

Reply:
also i feel `origin/main` is too hard coded, maybe use something like this to get the base branch of the PR, then do a diff

https://github.com/joshuali925/Jest-Coverage-Diff/blob/ce71109d70ec4e2103b0ee9cf261eb3953ad7a7a/src/main.ts#L25